### ISSUE TITLE
Registered Type Adapters not being used

### ISSUE THREAD

**Issue by WillEllis** [square/retrofit#1419]
Hi,

I'm attempting to use Retrofit 2.0 for my service calls. The JSON I am receiving can mostly be handled correctly by all the default GSON type adapters. But some of the JSON has WCF .Net DateTime values. 
I have handled these successfully before with just OkHTTP and GSON by registering a type adapter like so:
`Gson gson = new GsonBuilder()
                    .registerTypeAdapter(Calendar.class, new DotNetDateTimeAdapter())
                    .create();`

Now that I am using Retrofit I realised I needed to use GsonConverterFactory passing in the gson instance:
`Retrofit retrofit = builder
                .baseUrl(serverAddress)
                .client(httpClient)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();`

I find now I get exceptions like:
`com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 149 path $[0].CreationDate`

And looking at the stack trace it appears that the default GSON typeadapter for calendar is being used.
Surely if I have registered a custom type adapter with Gson and passed this to Retrofit my own custom adapter should be used. Am I using GsonConverterFactory incorrectly?

**Comment by JakeWharton** [comment]
We have a test that asserts a type adapter works: https://github.com/square/retrofit/blob/90729eb2ae2f3329281c1f9813ab1de3daa71ad0/retrofit-converters/gson/src/test/java/retrofit2/GsonConverterFactoryTest.java.

Can you phrase your bug in the form of a failing test?

**Comment by WillEllis** [comment]
I've realised the way I was using the retrofit builder meant I'd ended up in my code setting the converter factory in two places, so first it was registering the default gson converter and then later registering the custom one.
It seems the second call is ignored. 
I've corrected this so it only sets once and now it works, my bad! 
Thanks for pointing me in the direction of the unit tests though, they seems to be a great resource for usage of retrofit 2.0. Loving the library! Cheers
